### PR TITLE
Cleanup logging and naming

### PR DESCRIPTION
**Pull request by Shadowghost**
This bothered me for way too long:
* Only log something if something actually changed
* Omit logging item ids if debugging isn't enabled
* Make log messages more clear
* Fix the scheduled task section name
* Upgrades LiteDB to v5, fixes incompatibilities and adds `upgrade=true` to connection string to automatically upgrade the database

### CODE REVIEW

File: Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs [hunk 1]
```diff
@@ -156,8 +155,20 @@ private void SaveUserChanges(List<MediaBrowser.Model.Dto.UserItemDataDto> dtos,
...
+            {
+                _logger.LogInformation(
+                    "User Data Sync: User {UserId}({Username}) posted {NumberOfUpdates} updates",
+                    userId,
+                    userName,
+                    itemCount);
+
+                _logger.LogDebug(
+                    "Updated items: {Updates}",
+                    string.Join(",", ids.ToArray()));
```
Comment (crobibero) [inline Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs:170]: You can just pass the list of IDs

https://github.com/serilog/serilog/wiki/Structured-Data#collections

If this doesn't log the desired outcome you can still remove the `ToArray` call

Comment (Shadowghost) [reply Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs:170]: I just used what was already there. Will test and change according to what's actually logged.

File: Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs [hunk 2]
```diff
@@ -156,8 +155,20 @@ private void SaveUserChanges(List<MediaBrowser.Model.Dto.UserItemDataDto> dtos,
...
+            {
+                _logger.LogInformation(
+                    "User Data Sync: User {UserName} ({UserId}) posted {NumberOfUpdates} updates",
+                    userName,
+                    userId,
+                    itemCount);
+
+                _logger.LogDebug(
+                    "Updated items: {Updates}",
+                    ids.ToArray());
```
Comment (crobibero) [inline Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs:170]:
```suggestion
                    ids);
```
this is already a list

File: Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs [hunk 3]
```diff
@@ -133,7 +132,7 @@ private void SendNotifications(IEnumerable<KeyValuePair<Guid, List<BaseItem>>> c
             {
                 cancellationToken.ThrowIfCancellationRequested();
                 var userId = pair.Key;
-                _logger.LogDebug("Starting to save items for {userId}", userId.ToString());
+                _logger.LogDebug("Started saving items for {userId}", userId.ToString());
```
Comment (crobibero) [inline Jellyfin.Plugin.KodiSyncQueue/EntryPoints/UserSyncNotification.cs:135]:
```suggestion
                _logger.LogDebug("Started saving items for {userId}", userId);
```

File: Jellyfin.Plugin.KodiSyncQueue/EntryPoints/LibrarySyncNotification.cs [hunk 4]
```diff
@@ -202,12 +201,19 @@ private void PushChangesToDb(IReadOnlyCollection<LibItem> itemsAdded, IReadOnlyC
...
+            if (itemCount > 0)
+            {
+                _logger.LogInformation(
+                    "Library Sync: {StatusType} {NumberOfItems} items",
+                    status,
+                    items.Count);
+
+                _logger.LogDebug(
+                    "Affected items: {Items}",
+                    string.Join(",", items.Select(i => i.Id.ToString("N", CultureInfo.InvariantCulture)).ToArray()));
```
Comment (crobibero) [inline Jellyfin.Plugin.KodiSyncQueue/EntryPoints/LibrarySyncNotification.cs:215]:
```suggestion
                    items.Select(i => i.Id.ToString("N", CultureInfo.InvariantCulture)));
```

**Review by crobibero — COMMENTED** [review]
I know these already existed, but might as well clean them up while cleaning up logging

**Review by crobibero — APPROVED** [review]

**Pull request merged**